content/articles/line-detection-using-hough-transform-in-matlab/index.md

```diff
@@ -0,0 +1,149 @@
+
+Now, we use a simple image to implement the above algorithm. We have two binary images shown below;
+
+![Binary images](/engineering-education/line-detection-using-hough-transform-in-matlab/image-six.png)
+
+These images are binary because hough transform function works on binary images. Note that it is possible to plot these two images in Matlab. We will first plot these two images and extract line segments. We first plot the image with the two dots and extract lines from it before going to the second.
+To plot the image with the two dots, we use execute the code below;
+```matlab
+close all
+clear
+clc
+%% 1. Image with two dots
+a = zeros(50);
+a(20,20) = 1; a(40,10) = 1;
+figure;
+subplot(2,2,1)
+imshow(a)
+title('Image with 2 dots')
+```
+`Close all` is used to close all the open figures, and `clc` clears the command window. We then make a matrix of zeros of the dimension 50x50 using `zero(50)`. `a(20, 20)` and `a(40, 10)` are the points at which the dots are placed. `Figure` creates a figure window. We then use the `imshow()` to view the output. Finally, using the `subplot()` function, we create a subplot for our output. `subplot(m,n,p)` creates an `MxN` axes, and the plot is placed at position `p`.
+We use the `hough()` function to compute the hough matrix. This function takes in binary images as the input. This function gives out three outputs, that is, Transform matrix(H), theta(T), and rho(R). After that, we visualize the output using the utility function `houghMatViz()` as shown below:
+```Matlab
+[H, T, R] = hough(a);
+subplot(2, 2, 2)
+houghMatViz(H, T, R)
+```
+>Let us look at the content of this utility function. Remember, this function should be in a different script, and then we call it.
+```Matlab
+function houghMatViz(H,T,R)
+Hgray = mat2gray(H); % Converting hough matrix(H) into grayscale image.
+Hgray = imadjust(Hgray); % Enhancing the brightness
+
+imshow(Hgray,'XData',T,'YData',R);
+hold on
+
+axis on % turning on the axis.
+axis normal 
+colormap(hot);title('Hough Transform'); % Giving colormap to the image.
+
+xlabel('\theta')  % add x-y labels
+ylabel('\rho');
+```
+This utility function converts the hough matrix into a grayscale image using the `mat2gray()` function. Then, the brightness of the grayscale image is adjusted using the `imadjust()` function. Finally, this function takes the grayscale image as the argument. `imshow()` displays the output depending on the `XData` and the `YData`.
+
+When we execute this code, we get two sinusoids that correspond to the two points in the image, as shown below:
+
+![Two sinusoids corresponding to the two points](/engineering-education/line-detection-using-hough-transform-in-matlab/image-seven.png)
+
+We use the `houghpeaks()` function to find the peaks. Using this function, you can provide the number of peaks required to be returned as the output. The output will be the largest peak if you fail to provide this.
```

Comment:
` The output will be the largest peak if you fail to provide this.`

Elaborate this sentence.